src/components/DetectActive/DetectActive.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import React, { Component } from 'react';
+
+type Props = {
+  children: (isActive: boolean) => React$Node,
+};
+
+type State = {
+  isActive: boolean,
+};
+
+class DetectActive extends Component<Props, State> {
```

Comment:
Both `StrokeButton` and `SelectableItem` needed the same hover behaviours, and because events propagate, I was able to create its own component for it.